ui/src/views/Settings.vue

```diff
@@ -1,63 +1,103 @@
+<!--
+  Copyright (C) 2022 Nethesis S.r.l.
+  SPDX-License-Identifier: GPL-3.0-or-later
+-->
 <template>
-  <div class="bx--grid bx--grid--full-width">
-    <div class="bx--row">
-      <div class="bx--col-lg-16 page-title">
+  <cv-grid fullWidth>
+    <cv-row>
+      <cv-column class="page-title">
         <h2>{{ $t("settings.title") }}</h2>
-      </div>
-    </div>
-    <!-- sample settings page -->
-    <!-- <div v-if="error.getConfiguration" class="bx--row">
-      <div class="bx--col">
+      </cv-column>
+    </cv-row>
+    <cv-row v-if="error.getConfiguration">
+      <cv-column>
         <NsInlineNotification
           kind="error"
           :title="$t('action.get-configuration')"
           :description="error.getConfiguration"
           :showCloseButton="false"
         />
-      </div>
-    </div>
-    <div class="bx--row">
-      <div class="bx--col-lg-16">
-        <cv-tile :light="true">
-          <cv-form @submit.prevent="saveSettings"> -->
-    <!-- TODO configuration fields -->
-    <!-- <div v-if="error.configureModule" class="bx--row">
-              <div class="bx--col">
+      </cv-column>
+    </cv-row>
+    <cv-row>
+      <cv-column>
+        <cv-tile light>
+          <cv-form @submit.prevent="configureModule">
+            <cv-text-input
+              :label="$t('settings.nethvoice_host')"
+              v-model="form.nethvoice_host"
+              placeholder="voice.example.com"
+              :disabled="loading.getConfiguration || loading.configureModule"
+              :invalid-message="error.nethvoice_host"
+              ref="nethvoice_host"
+            />
+            <cv-text-input
+              :label="$t('settings.nethcti_ui_host')"
+              v-model="form.nethcti_ui_host"
+              placeholder="cti.example.com"
+              :disabled="loading.getConfiguration || loading.configureModule"
+              :invalid-message="error.nethcti_ui_host"
+              ref="nethcti_ui_host"
+            />
+            <cv-toggle
+              :label="$t('settings.lets_encrypt')"
+              value="lets_encrypt"
+              :disabled="loading.getConfiguration || loading.configureModule"
+              v-model="form.lets_encrypt"
+            />
+            <!-- TODO: use cluster/list-user-domains to list available domains, otherwise allow manual imput --->
```

Comment:
Let's remove the TODO and implement it.

Reply:
Will take a bit of time, due to complexity of the component, but I'm on it